### DISCUSSION TITLE
is there a canonical way to refer to content in an image?

### DISCUSSION THREAD

**Discussion by beav** [bootc-dev/bootc#254] [Q&A] [answered]
Looking at the bootc status output, I see `<hostname>:<port>/<path>/<image-name>:<tag>` as the spec for referring to the installed image. However, the tag can point to different content over time as new things are pushed to the registry.

What would be the correct way to canonically refer to particular content from a image? would it be `<image-name>:<imageDigest>`?

**Comment by cgwalters** [comment, marked as answer]
Right, you can just pass a digested pull spec to `bootc install/bootc switch` if you want; then the image is pinned unless explicitly unpinned after an explicit `bootc switch`.

This by the way is effectively how OpenShift nodes work by default today.

**Reply by cgwalters** [reply]
> What would be the correct way to canonically refer to particular content from a image? would it be <image-name>:<imageDigest>?

And to clarify yes - this is the same way all other OCI containers work, you can use tags or you can use digest.

There is some support for tag *and* digest, but...this has some slightly weird semantics.
